core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -906,10 +906,10 @@ private[spark] class SparkSubmit extends Logging {
 
     if (args.verbose) {
       logInfo(s"Main class:\n$childMainClass")
-      logInfo(s"Arguments:\n${childArgs.mkString("\n")}")
+      logInfo(s"Arguments:\n${childArgs.sorted.mkString("\n")}")
       // sysProps may contain sensitive information, so redact before printing
-      logInfo(s"Spark config:\n${Utils.redact(sparkConf.getAll.toMap).mkString("\n")}")
-      logInfo(s"Classpath elements:\n${childClasspath.mkString("\n")}")
+      logInfo(s"Spark config:\n${Utils.redact(sparkConf.getAll.toMap).sorted.mkString("\n")}")
+      logInfo(s"Classpath elements:\n${childClasspath.sorted.mkString("\n")}")
```

Comment:
Likewise classpath order matters, so not sure we should do this.
Spark configs ... yeah order should not matter, so that should be sorted.

Reply:
Reverted it!